src/datetime.ts

```diff
@@ -52,7 +52,7 @@ export interface DateTime {
   month?: Month | string;
 
   /**
-   * Integer value representing the date from 1-31.
+   * Integer value representing the date (day of the month) from 1-31.
```

Comment:
By definition, dates mean days of month. Plus, we already say 1-31.  So these parentheses are a bit overkill.

Reply:
I appreciate the additional clarification since JavaScript uses `day` for the day of the month. With the clarification, you can ctrl+f for `day` and find the right name. Also, date is not generally understood as the day fo the month. Where did you see this?

Reply:
> since JavaScript uses day for the day of the month.

No, it does not. We do match JS. Day is day of week, as it should be. 

![image](https://user-images.githubusercontent.com/111269/73129470-0682c100-3f99-11ea-9b2a-14001bd0dd2c.png)


> Also, date is not generally understood as the day fo the month. Where did you see this?

This is well defined in the dictionary. 

![image](https://user-images.githubusercontent.com/111269/73129478-2619e980-3f99-11ea-914f-89d59f9a996c.png)

Reply:
I agree, I don't think the additional clarification hurts.

Reply:
> No, it does not. We do match JS. Day is day of week, as it should be.

Oh, that's confusing. I missed that `getDate` returns the day of the month but the parameter to the constructor is called `day`. 🤦‍♂ 

<img width="1036" alt="Screen Shot 2020-01-26 at 10 05 32" src="https://user-images.githubusercontent.com/589034/73139498-6cb42600-4023-11ea-8f6d-d2a8598a4963.png">

> This is well defined in the dictionary.

I didn't find https://en.wikipedia.org/wiki/Calendar_date clarified this well. I asked @ldegreef what she thinks a "date" was and she thought that date refers to the whole time specifier (like here: https://en.wikipedia.org/wiki/ISO_8601#Dates). Moreover, in this pull request, I replaced a few occurrences of `day` with `date` since were assigned numbers above 7, which indicates that we are confused ourselves about what `day` and `date` mean. Of course, it's too late, but maybe `dayOfMonth` and `dayOfWeek` or `date` and `weekday` would have been better names to avoid confusion.